crates/transaction-pool/src/pool/listener.rs

```diff
@@ -64,28 +65,42 @@ impl Stream for AllTransactionsEvents {
 ///
 /// This is essentially a multi-producer, multi-consumer channel where each event is broadcast to
 /// all active receivers.
-#[derive(Default, Debug)]
-pub(crate) struct PoolEventBroadcast {
+#[derive(Debug)]
+pub(crate) struct PoolEventBroadcast<T: PoolTransaction> {
     /// All listeners for all transaction events.
-    all_events_broadcaster: AllPoolEventsBroadcaster,
+    all_events_broadcaster: AllPoolEventsBroadcaster<T>,
     /// All listeners for events for a certain transaction hash.
     broadcasters_by_hash: HashMap<TxHash, PoolEventBroadcaster>,
 }
 
-impl PoolEventBroadcast {
+impl<T: PoolTransaction> Default for PoolEventBroadcast<T> {
+    fn default() -> Self {
+        Self {
+            all_events_broadcaster: AllPoolEventsBroadcaster::default(),
+            broadcasters_by_hash: HashMap::default(),
+        }
+    }
+}
+
+impl<T: PoolTransaction> PoolEventBroadcast<T> {
     /// Calls the broadcast callback with the `PoolEventBroadcaster` that belongs to the hash.
-    fn broadcast_event(&mut self, hash: &TxHash, event: TransactionEvent) {
+    fn broadcast_event(
+        &mut self,
+        hash: &TxHash,
+        event: TransactionEvent,
+        pool_event: PoolTransactionEvent<T>,
```

Comment:
this could be a bit confusing at first, so I think we should add a few docs how these differ to `PoolTransactionEvent` and `TransactionEvent` 

still not quite happy with the name, perhaps we rename `PoolTransactionEvent` to `FullTransactionEvent` ?